src/language.h

```diff
@@ -155,30 +117,48 @@ struct StringParameters {
 
 	bool set_mode; ///< When not in set-mode, all parameters are cleared on first use of a Set function.
 	std::vector<StringParameterData> parms; ///< Parameters of the string.
+	int64 pluralize_count;                  ///< Value to use for selecting the plural form.
 };
 
-/**
- * Class for retrieving language strings.
- * @todo Implement me.
- */
-class Language {
+/** A string in one language with all its plural forms. */
+using PluralizedString = std::vector<const char*>;
+
+/** Contains all strings and the meta-information for one specific language. */
+class LanguageBundle {
+public:
+	LanguageBundle();
+	void Clear();
+	void InitMetaInfo(int index);
+
+	const char *GetSgText(StringID number) const;
+	const char *GetPlural(StringID number, int64 count) const;
+
+	std::vector<PluralizedString> values;  ///< Every known string with all its plural forms, indexed by #StringID.
```

Comment:
Vectors may pre-allocate more entries than you use (to reduce the number reallocations).
For the outer vector this should not be a problem.

For the inner vector, you may want to do something. If you `reserve` for all plural forms (before adding elements), the vector should not grow any more.

Another solution could be an `std::array` as inner container. You get the precise number of entries at minimal space usage then, but the `Language` class becomes templated with the number of plurals (or, I am guessing, the plural form identification that you can use to obtain the number of plurals).  
Not sure that would work.

Another option is to flatten the inner vector into the outer vector by making it have space for all plural forms of each string. To get the right spot you'd need to multiply the string number by the number of plural forms.

Reply:
I can see you may want to postpone this until you have more languages.

Reply:
Actually the vector will sometimes contain `nplurals` elements and sometimes only 1 element, depending on whether the string in question does have plural forms. So an array and/or templates wouldn't work at all here. However I added a call to `shrink_to_fit` on the inner vectors.